turbinia/state_manager.py

```diff
@@ -328,3 +332,146 @@ def write_new_task(self, task):
+
+  def get_evidence(self, evidence_id: str):
+    """Gets one evidence from Redis given its ID.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+
+    Returns:
+      evidence_dict (dict): Dict containing evidence attributes. 
+    """
+    key = ':'.join(('TurbiniaEvidence', evidence_id))
+    evidence_keys = self.client.hkeys(key)
+    evidence_dict = {}
+    for attribute_key in evidence_keys:
+      evidence_dict[attribute_key.decode()] = json.loads(
+          self.client.hget(key, attribute_key))
+    return evidence_dict
+
+  def evidence_exists(self, evidence_id):
+    """Checks if the evidence is saved in Redis given its ID.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+
+    Returns:
+      evidence_id (bool): Boolean indicating if evidence is saved. 
+    """
+    return self.client.exists(':'.join(('TurbiniaEvidence', evidence_id)))
+
+  def get_evidence_summary(self, sort: str = None, output: str = 'keys'):
+    """Gets a summary of all evidences.
+
+    Args:
+      sort (str): Name of the evidence attribute used to sort the output.
+      output (str): Value to be output by the function (keys | values | count).
+
+    Returns:
+      summary (dict | int): Object containing evidences. 
+    """
+    if output == 'count' and not sort:
+      return sum(1 for _ in self.client.scan_iter('TurbiniaEvidence:*'))
+    summary = {} if sort else []
+    for key in self.client.scan_iter('TurbiniaEvidence:*'):
+      value = self.get_evidence(key.decode().split(':')[1])
+      stored_value = value if output == 'values' else key.decode()
+      if sort:
+        attribute_value = value.get(sort, None)
+        if attribute_value not in summary:
+          summary[attribute_value] = [stored_value] if output != 'count' else 1
+        elif output == 'count':
+          summary[attribute_value] += 1
+        else:
+          summary[attribute_value].append(stored_value)
+        continue
+      summary.append(stored_value)
+    return summary
+
+  def query_evidence(
+      self, attribute_name: str, value: Any, output: str = 'keys'):
+    """Queries for evidences with the specified attribute matching the passed
+    value.
+
+    Args:
+      attribute_name (str): Name of the attribute containing the value.
+      value (Any): Value stored in the attribute.
+      output (str): Value to be output by the function (keys | values | count).
+
+    Returns:
+      summary (dict): Dict containing evidences of each type. 
+    """
+    keys = []
+    for key in self.client.scan_iter('TurbiniaEvidence:*'):
+      if stored_value := self.client.hget(key, attribute_name):
+        if (attribute_name == 'tasks' and
+            value in json.loads(stored_value)) or stored_value.decode(
+            ) == value or json.loads(stored_value) == value:
+          keys.append(key.decode())
+    if output == 'values':
+      return [self.get_evidence(key.split(':')[1]) for key in keys]
```

Comment:
Do we need to call key.decode() before calling split() here like you do in like 446?

Reply:
Those keys were decoded before appending to "keys" in 446, so they are already strings